docs/instruments/ZJSONInstrument_v0.0.1.md

```diff
@@ -0,0 +1,1028 @@
+- [date](#512-date-element)
+- [number](#513-numeric-element)
+- [radio](#514-radio-button-element)
+- [select](#515-select-element)
+- [text](#516-text-element)
+- [textarea](#517-textarea-element)
+- [time](#518-time-element)
+
+For helpers:
+
+- [header](#519-header-element) (from h1 to h6)
+- [link](#5110-link-element)
+- [static](#5111-static-element)
+
+These are all mappable to some LORIS React Form element, except for "header" which is native to HTML.
+
+### 5.1.1 Checkbox Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "checkbox"
+        }
+    }
+}
+```
+
+### 5.1.2 Date Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "date"
+        }
+    }
+}
+```
+
+### 5.1.3 Numeric Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "number"
+        }
+    }
+}
+```
+
+### 5.1.4 Radio Button Element
+
+```js
+{
+    "ui": {
+        ...
+        "nameOfRadioButtons" : {
+            "type" : "radio",
+            "options" : {
+                "vertical" : boolean
+            }
+        }
+    }
+}
+```
+
+`options.vertical`: Boolean. True or false that the radio group is rendered vertically. The default value is false, or
+horizontally. 
+
+### 5.1.5 Select Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "select"
+        }
+    }
+}
+```
+
+### 5.1.6 Text Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "text",
+            "options": {
+                "placeholder": string
+            }
+        }
+    }
+}
+```
+
+`options.placeholder`: String. Text that is rendered inside the text element as a placeholder before user input.
+
+### 5.1.7 Textarea Element
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "textarea",
+            "options": {
+                "placeholder": string
+            }
+        }
+    }
+}
+```
+
+`options.placeholder`: String. Text that is rendered inside the text element as a placeholder before user input.
+
+### 5.1.8 Time Element 
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "time"
+        }
+    }
+}
+```
+
+### 5.1.9 Header Element
+
+A header element renders a "static" `helper` that should be displayed with some level of prominence.
+
+```js
+{
+    "ui": {
+        "nameOfField" : {
+            "type" : "header",
+            "options": {
+                "level": int
+            }
+        }
+    }
+}
+```
+
+`options.level`: Int. The number indicating the level of the html heading element to be rendered i.e. h1, the highest
+and most prominent, and h6, the lowest and reserved for subheadings.
+
+### 5.1.10 Link Element
+
+```js
+{
+    "ui": {
+        "nameOfHelper" : {
+            "type" : "link",
+        }
+    }
+}
+```
+
+### 5.1.11 Static Element
+
+```js
+{
+    "ui": {
+        "nameOfHelper" : {
+            "type" : "static"
+        }
+    }
+}
+```
```

Comment:
Still not sure what a "static element" means in this spec..

It would probably be a good idea to include an "image" type as well.

Reply:
And probably File as well ?

Reply:
I think files would be field types and not UI types, but that's a good idea..

Reply:
file will be for second phase implementation. i like implementation a ui image type but will need to think how that helper element would be defined